### ISSUE TITLE
Docs: Skip to Content on homepage shouldn't skip the first section of main content

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#5212]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Right now, if you <kbd>Tab</kbd> to the _Skip to Content_ button on the homepage and hit <kbd>Enter</kbd> to activate it, the page scrolls down to the _What are ESLint and TypeScript, and how do they compare?_ heading. That means users skip the entire _TypeScript ESLint_ & _Tooling which enables ESLint to support TypeScript_ section. 

![Screenshot of the Skip to Content button on the homepage](https://user-images.githubusercontent.com/3335181/174569193-1f6281a8-6b2f-4c29-9b86-f8bb39dfe27b.png)

Users should be scrolled+focused to the _TypeScript ESLint_ section since it's the first main content in the page.

Discussed on stream with @BenDMyers on Twitch: https://www.twitch.tv/videos/1498962341. See https://benmyers.dev/blog/skip-links for a description of skip link issues.

### Affected URL(s)

https://typescript-eslint.io

**Comment by Josh-Cena** [comment]
Docusaurus' STC button will select `main:first-of-type` to focus. The banner is a `header` element, not `main`, so to focus it we have to make the title a part of `main` as well—not sure how good that idea is, considering I'd consider "Tooling which enables ESLint to support TypeScript" more decorative than informative.

**Comment by JoshuaKGoldberg** [comment]
Yeah I was wondering about that too. The things that tipped us over into considering the banner to be main content were:

* The banner is unique to the homepage
* That _Tooling which ..._ phrase is not repeated for a while on the page
* The two buttons don't appear for a little while on the page -- especially _Playground_

I'll also add that it takes up a good portion of the screen real estate, especially on mobile/tablet resolutions.

**Comment by Josh-Cena** [comment]
Related: https://github.com/facebook/docusaurus/issues/6411 I wonder if we should focus to a `#main-content` ID in priority... This would fix that issue as well (we can make an anchor-like redirect and then erase the fragment, like GitHub does)

**Comment by kemilbeltre** [comment]
> Related: [facebook/docusaurus#6411](https://github.com/facebook/docusaurus/issues/6411) I wonder if we should focus to a `#main-content` ID in priority... This would fix that issue as well (we can make an anchor-like redirect and then erase the fragment, like GitHub does)

@Josh-Cena @JoshuaKGoldberg  are you still accepting PRs?

**Comment by JoshuaKGoldberg** [comment]
Yup, go ahead - please do! ❤️

**Comment by kemilbeltre** [comment]
@JoshuaKGoldberg Looking at the current implementation of docusaurus, the solution is to move the header section to the main content.

I'll leave a PR with the changes. Thanks
